packages/components/src/FormField/FormField.tsx

```diff
@@ -95,7 +105,7 @@ export function FormField(props: FormFieldProps) {
 
   const fieldProps = {
     ...rest,
-    id: identifier,
+    id,
     className: styles.input,
     name: (validations || name) && controllerName,
```

Comment:
Only a nit, we seem to have `name`, `controlledName`, and `controllerName`. Just wondering if simplifying all these names would make it easier. I don't even think we need to extract `controllerName` from `useController` since we're spreading `...rest` onto `fieldProps`. So there's a few optimizations, but we may just want to rename `name: nameProp` in the destructuring, then we can use:
```
const name = nameProp ?? `generatedName--${id}`
```

Then we can replace this here with:
```suggestion
    name: (validations || nameProp) && name,
```

Reply:
Fair enough. Seemed like only 3 places so I figured it was pretty lo-risk since it keeps most of the logic contained rather than referring to different values as it goes down the file which actually makes it more bug prone.